### PR TITLE
chore(e2e-tests): setup insurance tests in seperate directory

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Changes made:
* Split e2e directory into quote, insurance and switch-between-services
* Shared constants between all
* Rewritten test.yml to split workflows for tests

### CODE REVIEW

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Please follow same approach as `https://github.com/UK-Export-Finance/dtfs2/blob/main/.github/workflows/test.yml`

File: e2e-tests/insurance/cypress/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      object: true,
+    }],
+  },
+  parserOptions: {
+    ecmaVersion: 12, // es2021
+  },
+  globals: {
+    cy: true,
+  },
+};
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/.eslintrc.js:42]: Do we need to have the eslint config duplicate in `insurance/.eslintrc.js` and `insurance/cypress.eslintrc.js`?

File: e2e-tests/quote/cypress/support/commands.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+Cypress.Commands.add('checkValue', require('../../../commands/check-value'));
+Cypress.Commands.add('checkAriaLabel', require('../../../commands/check-aria-label'));
+Cypress.Commands.add('checkRadioInputYesAriaLabel', require('../../../commands/check-radio-input-yes-aria-label'));
+Cypress.Commands.add('checkRadioInputNoAriaLabel', require('../../../commands/check-radio-input-no-aria-label'));
+
+Cypress.Commands.add('checkTaskStatus', require('../../../commands/check-task-status'));
+Cypress.Commands.add('checkTaskStatusCompleted', require('../../../commands/check-completed-task-status'));
+Cypress.Commands.add('checkLink', require('../../../commands/check-link'));
+Cypress.Commands.add('getReferenceNumber', require('../../../commands/get-reference-number'));
+Cypress.Commands.add('keyboardInput', require('../../../commands/keyboard-input'));
```
Comment (ttbarnes) [inline e2e-tests/quote/cypress/support/commands.js:145]: Commands seem to be duplicated in the insurance and quote directories. Are we able to be DRY?

Comment (Zainzzkk) [reply e2e-tests/quote/cypress/support/commands.js:41]: done

File: e2e-tests/insurance/cypress/.eslintrc.js (hunk 1 above)
Comment (Zainzzkk) [reply e2e-tests/insurance/cypress/.eslintrc.js:42]: done

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -11,7 +11,7 @@ on:
   pull_request:
     branches: [main-application]
     paths:
-      - '.github/workflows/test.yml'
+      - '.github/workflows/**'
```
Comment (abhi-markan) [inline .github/workflows/test.yml:14]: Can we keep to `test.yml` we don't want to run this pipeline for IaC deployment.

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Few comments to address please.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -71,6 +102,32 @@ jobs:
...
+        uses: actions/checkout@v3
+
+      - name: Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ vars.NODE_VERSION }}
+
       - name: API
         working-directory: ./
         env:
```
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply .github/workflows/test.yml:133]: Spoke to @abhi-markan.  Keystone fails to build without these

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -102,13 +159,71 @@ jobs:
...
-    name: E2E 👷
+  # 4. E2E - Insurance
+  e2e-tests-insurance:
+    name: E2E Insurance 👷
+    needs: setup
+    environment:
+      name: ${{ needs.setup.outputs.environment }}
+    runs-on: ubuntu-latest
+
+    strategy:
```
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply .github/workflows/test.yml:170]: Spoke to @abhi-markan .  concurrency not working for e2e workflows

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -20,6 +20,32 @@ on:
 env:
   environment: 'qa'
```
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply .github/workflows/test.yml:21]: changed to single quotes

File: .github/workflows/test.yml (hunk 3 above)
Comment (Zainzzkk) [reply .github/workflows/test.yml:14]: changed

File: e2e-tests/insurance/cypress/.env.sample [hunk 7]
```diff
@@ -0,0 +1,8 @@
+# NODE
+NODE_ENV=
```
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply e2e-tests/insurance/cypress/.env.sample:2]: added API_KEY

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**